<a href="https://colab.research.google.com/github/pranita30012005/ChronicDiseaseRiskAnalysis/blob/main/Day_1_%E2%80%94_Notebook_1_Project_Setup.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 🩺 Chronic Disease Risk Factor Analysis
### Day 1 — Notebook 1: Project Setup
**Goal:** Set up our working environment, load the Pima Indians Diabetes dataset,
and take a first look at its structure.

In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

print("Libraries imported successfully!")
print("Pandas version:", pd.__version__)
print("NumPy version:", np.__version__)

Libraries imported successfully!
Pandas version: 2.2.2
NumPy version: 2.0.2


In [5]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [15]:
DATA_PATH = "/content/drive/MyDrive/ChronicDiseaseRiskAnalysis/data/raw/diabetes.csv" # Updated path based on Google Drive listing
df = pd.read_csv(DATA_PATH)

print("Dataset loaded successfully!")
print("Shape:", df.shape)

Dataset loaded successfully!
Shape: (768, 9)


In [13]:
!ls -R /content/drive/MyDrive

/content/drive/MyDrive:
 ChronicDiseaseRiskAnalysis  'Colab Notebooks'

/content/drive/MyDrive/ChronicDiseaseRiskAnalysis:
data  notebooks

/content/drive/MyDrive/ChronicDiseaseRiskAnalysis/data:
processed  raw

/content/drive/MyDrive/ChronicDiseaseRiskAnalysis/data/processed:

/content/drive/MyDrive/ChronicDiseaseRiskAnalysis/data/raw:
diabetes.csv

/content/drive/MyDrive/ChronicDiseaseRiskAnalysis/notebooks:

'/content/drive/MyDrive/Colab Notebooks':
 01_Project_Setup.ipynb			        Untitled0.ipynb
'Day 2 — Data Cleaning & Preprocessing.ipynb'


In [16]:
df.head()

,Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age,Outcome
0,6,148,72,35,0,33.6,0.627,50,1
1,1,85,66,29,0,26.6,0.351,31,0
2,8,183,64,0,0,23.3,0.672,32,1
3,1,89,66,23,94,28.1,0.167,21,0
4,0,137,40,35,168,43.1,2.288,33,1


In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [18]:
df.describe()

,Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age,Outcome
count,768.000000,768.000000,768.000000,768.000000,768.000000,768.000000,768.000000,768.000000,768.000000
mean,3.845052,120.894531,69.105469,20.536458,79.799479,31.992578,0.471876,33.240885,0.348958
std,3.369578,31.972618,19.355807,15.952218,115.244002,7.884160,0.331329,11.760232,0.476951
min,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.078000,21.000000,0.000000
25%,1.000000,99.000000,62.000000,0.000000,0.000000,27.300000,0.243750,24.000000,0.000000
50%,3.000000,117.000000,72.000000,23.000000,30.500000,32.000000,0.372500,29.000000,0.000000
75%,6.000000,140.250000,80.000000,32.000000,127.250000,36.600000,0.626250,41.000000,1.000000
max,17.000000,199.000000,122.000000,99.000000,846.000000,67.100000,2.420000,81.000000,1.000000


In [19]:
df['Outcome'].value_counts()

,count
Outcome,
0,500
1,268


## 📝 Day 1 Observations

- **Rows:** 768    ** Columns:** 9  

- **Suspicious values I noticed:**   *Several columns (Glucose, BloodPressure, SkinThickness, Insulin, and BMI) show a minimum value of 0. This is medically impossible for a living person and likely represents missing data mapped as zeros.*  

  **- Class balance (Outcome):** 65.1% non-diabetic, 34.9% diabetic  

  **- One question I still have about this data:** *How should we handle or impute these medically impossible zero values without corrupting the dataset's integrity?*